lib/Chat/Parser/SystemMessage.php

```diff
@@ -509,6 +509,13 @@ protected function parseMessage(Message $chatMessage): void {
 				$parsedMessage = $this->l->t('{actor} shared a file which is no longer available');
 				if ($currentUserIsActor) {
 					$parsedMessage = $this->l->t('You shared a file which is no longer available');
+				} elseif ($currentActorType === Attendee::ACTOR_FEDERATED_USERS) {
+					$parsedMessage = $this->l->t('File shares are currently not supported in federated conversations');
+					$metaData = $parameters['metaData'] ?? [];
+					if (isset($metaData['caption']) && $metaData['caption'] !== '') {
+						$parsedMessage = '*' . $parsedMessage . "*\n\n" . $metaData['caption'];
+						$chatMessage->setMessageType(ChatManager::VERB_MESSAGE);
```

Comment:
A bit inconsistent for me, that messages without caption are system (and centered), while others look like normal chat messages.

Maybe make both normal, and pass a file object (at least name, if we're afraid of data loss), and rest could be handled on frontend side:
![image](https://github.com/nextcloud/spreed/assets/93392545/8d9d1661-165f-4454-b82d-4d4593739b17)